test/types/type-provider.test-d.ts

```diff
@@ -134,13 +152,22 @@ expectAssignable(server.withTypeProvider<TypeBoxProvider>().withTypeProvider<Jso
           y: { type: 'string' },
           z: { type: 'boolean' }
         }
+      } as const,
+      response: {
+        200: { type: 'string' },
+        '3xx': { const: '123' }
       } as const
     }
   },
-  (req) => {
+  (req, res) => {
     expectType<number | undefined>(req.body.x)
     expectType<string | undefined>(req.body.y)
     expectType<boolean | undefined>(req.body.z)
+
+    // response type narrowing
+    expectAssignable<(payload: string) => unknown>(res.code(200).send)
+    expectAssignable<(payload:'abc') => unknown>(res.code(200).send)
+    expectNotAssignable<(payload: string) => unknown>(res.code(301).send)
```

Comment:
Just as simple as
```ts
server.get('/', {
  schema: {
    response: {
      200: {
        type: 'string'
      }
    }
  }
}, (_, reply) => {
  return res.code(200).send('abc')
})
```

When there is an error with the `expectError` wrapper. It should fail the test.

Reply:
Sorry for getting back late, I had a busy weekend. I am not sure I understand how that would be a valid test. If you want to show that the return won't give an error, shouldn't the assignable tests be used. The first test checks that the callback which returns the `Reply` is assignable to the default `RouteHandlerMethod` and the second checks that the `Reply` is a valid return type for the default handler method.

Reply:
A use-case means to be a test itself. You don't really need to assert anything.
If error on an use-case, then it is perfect.

assertion for the guide is good, but how can you tell your hand-written one is matching the automatic resolved one by `TypeScript`. Since `automatic resolved` would change over time.

Reply:
I see what you mean. You don't want to just check for type errors but actual runtime errors. I added the test.